src/upload-tests.proj

```diff
@@ -77,6 +77,13 @@
     <HelixArchLabel>$(ArchGroup)</HelixArchLabel>
     <HelixConfigLabel>$(ConfigurationGroup)</HelixConfigLabel>
     <MaxRetryCount>1</MaxRetryCount>
+    <!-- Unique container name per build and configuration, e.g., corefx-beta-12345-01-windowsnt-x64-netcoreapp-master -->
+    <!-- Azure Storage containers can only contain [a-z, 0-9, -] in their names and $(OSGroup) is "Windows_NT" on Windows... -->
+    <ProcessedOSGroup>$([System.Text.RegularExpressions.Regex]::Replace('$(OSGroup.ToLower())', "[^A-Za-z0-9]+", ""))</ProcessedOSGroup>
+    <ProcessedArchGroup>$([System.Text.RegularExpressions.Regex]::Replace('$(ArchGroup.ToLower())', "[^A-Za-z0-9]+", ""))</ProcessedArchGroup>
+    <ProcessedTargetGroup>$([System.Text.RegularExpressions.Regex]::Replace('$(TargetGroup.ToLower())', "[^A-Za-z0-9]+", ""))</ProcessedTargetGroup>
+    <ProcessedBranch Condition="'$(Branch)' != ''">$([System.Text.RegularExpressions.Regex]::Replace('$(Branch.ToLower())', "[^A-Za-z0-9]+", ""))</ProcessedBranch>
+    <ContainerName Condition="'$(ContainerName)' == '' And '$(IsOfficial)' == 'true' And '$(Branch)' != ''">corefx-$(PreReleaseLabel)-$(BuildNumberMajor)-$(BuildNumberMinor)-$(ProcessedOSGroup)-$(ProcessedArchGroup)-$(ProcessedTargetGroup)-$(ProcessedBranch)</ContainerName>
```

Comment:
nit: I think I would put the branch next to the repo name so `corefx-$(ProcessedBranch)`